extensions/http-curl/protocols/RESTSender.cpp

```diff
@@ -99,11 +99,21 @@ const C2Payload RESTSender::sendPayload(const std::string url, const Direction d
     callback->ptr = input.get();
     callback->pos = 0;
     client.set_request_method("POST");
+    if (!ssl_context_service_) {
+      auto generatedService = std::make_shared<minifi::controllers::SSLContextService>("Service", configuration_);
```

Comment:
Can we move the first two lines out of the direction branch to avoid duplication?